apps/studio/components/to-be-cleaned/Storage/StorageSettings/S3Connection.tsx

```diff
@@ -0,0 +1,337 @@
+            <Table
+              className=""
+              head={[
+                <Table.th key="">Description</Table.th>,
+                <Table.th key="">Access key</Table.th>,
+                <Table.th key="">Created at</Table.th>,
+                <Table.th key="actions" />,
+              ]}
+              body={
+                hasStorageCreds ? (
+                  storageCreds.data?.map((cred: any) => (
+                    <StorageCredItem
+                      key={cred.id}
+                      created_at={cred.created_at}
+                      access_key={cred.access_key}
+                      description={cred.description}
+                      id={cred.id}
+                      onDeleteClick={() => {
+                        setDeleteCredId(cred.id)
+                        setOpenDeleteDialog(true)
+                      }}
+                    />
+                  ))
+                ) : (
+                  <Table.tr>
+                    <Table.td colSpan={3}>
+                      <p className="text-sm text-foreground">No credentials created</p>
+                      <p className="text-sm text-foreground-light">
+                        There are no credentials associated with your project yet
+                      </p>
+                    </Table.td>
+                  </Table.tr>
+                )
+              }
+            />
+          </div>
+        )}
+      </div>
+      <Dialog open={openDeleteDialog} onOpenChange={setOpenDeleteDialog}>
+        <DialogContent className="p-4">
+          <DialogTitle>Revoke storage credentials</DialogTitle>
+          <DialogDescription>
+            This action will permanently revoke the credentials. Requests made with these
+            credentials will stop working. Are you sure you want to revoke these credentials?
+          </DialogDescription>
+          <div className="flex justify-end gap-2">
+            <Button
+              type="outline"
+              onClick={() => {
+                setOpenDeleteDialog(false)
+                setDeleteCredId(null)
+              }}
+            >
+              Cancel
+            </Button>
+            <Button
+              type="danger"
+              loading={deleteStorageCreds.isLoading}
+              onClick={async () => {
+                if (!deleteCredId) return
+                await deleteStorageCreds.mutateAsync({ id: deleteCredId })
+                setOpenDeleteDialog(false)
+                toast.success('Storage credentials revoked')
+              }}
+            >
+              Yes, revoke credentials
+            </Button>
+          </div>
+        </DialogContent>
+      </Dialog>
+    </div>
+  )
+}
+
+function StorageCredItem({
+  description,
+  id,
+  created_at,
+  access_key,
+  onDeleteClick,
+}: {
+  description: string
+  id: string
+  created_at: string
+  access_key: string
+  onDeleteClick: (id: string) => void
+}) {
+  function daysSince(date: string) {
```

Comment:
You can just use https://date-fns.org/v3.6.0/docs/differenceInDays, right?

Reply:
how did i not know about this